targets/csharp/make.js

```diff
@@ -262,36 +265,63 @@ function getAuthParams(apiCall, isApiInstance = false) {
         return "\"X-SecretKey\", PlayFabSettings.DeveloperSecretKey";
     if (apiCall.auth === "SecretKey" && isApiInstance === true)
         return "\"X-SecretKey\", developerSecretKey";
-
     else if (apiCall.auth === "SessionTicket")
-        return "\"X-Authorization\", PlayFabSettings.ClientSessionTicket";
+        return "\"X-Authorization\", clientSessionTicket";
     return "null, null";
 }
 
 function getRequestActions(tabbing, apiCall, isApiInstance = false) {
     if (apiCall.result === "LoginResult" || apiCall.request === "RegisterPlayFabUserRequest")
-        return tabbing + "request.TitleId = PlayFabSettings.TitleId ?? request.TitleId;\n"
+        return tabbing + "request.TitleId = request.TitleId ?? PlayFabSettings.TitleId;\n"
             + tabbing + "if (request.TitleId == null) throw new PlayFabException(PlayFabExceptionCode.TitleNotSet, \"Must be have PlayFabSettings.TitleId set to call this method\");\n";
     if (apiCall.auth === "EntityToken")
-        return tabbing + "if (PlayFabSettings.EntityToken == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n";
-    if (apiCall.auth === "SessionTicket")
-        return tabbing + "if (PlayFabSettings.ClientSessionTicket == null) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn, \"Must be logged in to call this method\");\n";
+        return tabbing + "#if !DISABLE_PLAYFABENTITY_API\n"
+	        + tabbing + "    if ((request.AuthenticationContext?.EntityToken ?? PlayFabSettings.EntityToken) == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n"
+			+ tabbing + "#else\n"    
+			+ tabbing + "    if ((PlayFabSettings.EntityToken) == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n"
+			+ tabbing + "#endif\n";
```

Comment:
Theoretically, it should not be required to check !DISABLE_PLAYFABENTITY_API on every entityToken-authenticated API method, because the entire API file should be wrapped properly.
If you're experiencing a problem that requires this, you should step back and try and figure out why the entire file isn't wrapped in !DISABLE_PLAYFABENTITY_API.